flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -484,11 +485,15 @@ public final void invoke() throws Exception {
 		}
 	}
 
+	protected boolean runMailboxStep() throws Exception {
+		return mailboxProcessor.runMailboxStep();
+	}
+
```

Comment:
I don't see an easy way how to do it, as we would have to provide multiple different test classes (one per each production `StreamTask` subclass?).

Reply:
I came up with this weird construction:
```
public class TestStreamTask<OUT, OP extends StreamOperator<OUT>> extends StreamTask<OUT, OP> {
	private final StreamTask<OUT, OP> delegatee;

	public TestStreamTask(StreamTask<OUT, OP> streamTask) {
		super(streamTask.getEnvironment());
		this.delegatee = streamTask;
	}

	public boolean runMailboxStep() throws Exception {
		return delegate.mailboxProcessor.runMailboxStep();
	}
        // init()
}
```
Then replace type `StreamTask` to `TestStreamTask` and create it as:
```
env -> new TestStreamTask<>(new MultipleInputStreamTask<>(env))
```
(should be a factory)

I think this is better than having "test" methods in `StreamTask`.

Exposing `mailboxProcessor` (via constructor?) is also an option. But IMO less preferable because it exposes more details.

Reply:
I don't see how your proposal should work @rkhachatryan ? After passing to the test harness
```
val streamTask = new TestStreamTask<>(new MultipleInputStreamTask<>(env))
```
every time test harness would access the `streamTask`, it would be accessing `TestStreamTask`, not `MultipleInputStreamTask`, unless you would override/delegate  all of the methods, which would be an unstable overkill.

I think better solution would be to inject, mailbox into the `StreamTask`, but that would require to create `[Source|TwoInput|MultipleInput]StreamTaskBuilder` classes (constructors of `StreamTask` are already messy), and it's also on an overkill for this (imo) small problem. I think I would have done, if not for the fact how much refactoring and clean up I'm already doing here and will be doing more in other places (`AbstractStreamOperator` :()

Reply:
Yes, we would have to delegate all methods.